application/models/thread.go

```diff
@@ -147,6 +139,39 @@ func (t *Thread) CreateWithParticipants(postUuid string, user User) error {
 	}
 
 	*t = thread
+	return t.ensureParticipants(post, user.ID)
+}
+
+func (t *Thread) ensureParticipants(post Post, userID int) error {
+	threadParticipants, err := t.GetParticipants([]string{})
+	if domain.IsOtherThanNoRows(err) {
+		err = errors.New("error getting threadParticipants for thread: " + err.Error())
+		return err
+	}
+
+	for _, uID := range []int{post.CreatedByID, userID} {
+		if err := t.createParticipantIfNeeded(threadParticipants, uID); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *Thread) createParticipantIfNeeded(tpUsers Users, userID int) error {
```

Comment:
`createParticipantIfNeeded` needs a test

Reply:
I pulled this code out of ensure Participants early on to make it more readable (using return statements, instead of extra `if`s.  It's not exported and gets tested by the tests for ensure Participants and CreateWithParticipants.

I think that should be good enough, don't you?

Reply:
In general, I'd advocate for testing the lower-level function (end of the call chain) more than the higher-level, because it requires fewer permutations to ensure coverage of all possible scenarios. In this case, there is only one caller to `createParticipantIfNeeded`, so I think your test coverage is 100%. Yes, it's good as it is.